storage/innobase/fil/fil0fil.cc

```diff
@@ -81,17 +81,11 @@ bool fil_space_t::try_to_close(fil_space_t *ignore_space, bool print_info)
   mysql_mutex_assert_owner(&fil_system.mutex);
   for (fil_space_t &space : fil_system.space_list)
   {
-    if (&space == ignore_space)
+    if (&space == ignore_space || space.is_being_imported())
       continue;
-    switch (space.purpose) {
-    case FIL_TYPE_TEMPORARY:
+    if (!fil_is_user_tablespace_id(space.id))
       continue;
-    case FIL_TYPE_IMPORT:
-      break;
-    case FIL_TYPE_TABLESPACE:
-      if (!fil_is_user_tablespace_id(space.id))
-        continue;
-    }
+    ut_ad(!space.is_temporary());
```

Comment:
I don’t remember my reasoning about this, and I see that this functional change is not mentioned in the commit message. I might have missed the fact that the `break` is inside a `switch` block. Maybe I thought that there was a premature `break` of the `for` loop that iterates `fil_system.space_list`.

In any case, it would seem to be counterproductive to close a file on which an extensive I/O bound operation is in progress (initially by `fil_tablespace_iterate()` invoking `PageConverter::run()`, finally by `buf_flush_list_space()`). I will mention this change in the commit message.

Sure, we can combine the two `if` conditions; it’s trivial short-circuit evaluation.